### ISSUE TITLE
CouchPotato Unrar Not Running

### ISSUE THREAD

**Issue by bigmascot** [RuudBurger/CouchPotatoServer#4542]
Hi All,

I'm running CouchPotato on a FreeNAS box and up until recently it has been perfect.  CP has just started flagging errors in its log saying that it cannot unpack movies as so suitable unpacker is insalled.  My system is a dedicated box with 8GB RAM, three 2TB disks in ZFS, and Core i5.

From memory, this has happened before, and after copying a new rar file from elsewhere (it could have been SickBeard but I couldn't say for sure) the problem was resolved - I can't remember what I did though!

I have looked at various similar issues via Google, but have been unable to find a fix.  There was one post which I think was on here suggesting that a users tries to run ./unrar from the appropriate CP path in a terminal.  I've tried that and it tells be that it can't run a binary file.

The issue is something of an annoyance and I'd be really grateful if anyone was willing to post any suggestions.

I have tried to unpack the movie manually via my windows machine and it works perfectly.

The error from the CP log reads as follows:

[tato.core.plugins.renamer] Failed to extract /media/downloads/xxxxxxxxxxxx.rar: No suitable RAR unpacker installed Traceback (most recent call last):
  File "/usr/pbi/couchpotato-amd64/share/couchpotato/CouchPotatoServer/couchpotato/core/plugins/renamer/main.py", line 1154, in extractFiles
    rar_handle = RarFile(archive['file'])
  File "/usr/pbi/couchpotato-amd64/share/couchpotato/CouchPotatoServer/libs/unrar2/__init__.py", line 102, in __init__
    RarFileImplementation.init(self, password)
  File "/usr/pbi/couchpotato-amd64/share/couchpotato/CouchPotatoServer/libs/unrar2/unix.py", line 66, in init
    stdoutdata, stderrdata = self.call('v', []).communicate()
  File "/usr/pbi/couchpotato-amd64/share/couchpotato/CouchPotatoServer/libs/unrar2/unix.py", line 96, in call
    return call_unrar([cmd] + soptions + ['--', self.archiveName] + files)
  File "/usr/pbi/couchpotato-amd64/share/couchpotato/CouchPotatoServer/libs/unrar2/unix.py", line 49, in call_unrar
    raise UnpackerNotInstalled("No suitable RAR unpacker installed")
UnpackerNotInstalled: No suitable RAR unpacker installed

Many thanks.

**Comment by CouchPotatoBot** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Fill in all the dots. They're there for a reason
- [ ] Post logs, either inline (for smaller logs) or using [Pastebin](http://pastebin.com/)
- [ ] Post full version information

Please make sure you also read [contribution guide](https://github.com/RuudBurger/CouchPotatoServer/blob/develop/contributing.md#issues) and followed all the steps. 
Make the title describe your issue. Having "CP not working" or "I get this bug" for 100 issues, isn't really helpful. My master will close issues if there isn't enough information. On a good day he will tag the issue on close with the reason (like `can't reproduce`), but usually he won't, the lazy asshat.

Sometimes my master seems like a grumpy cat and responds with short answers. This isn't (always) because he hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let him know. Maybe he can teach me his awesome ways.

Thanks!

**Comment by bigmascot** [comment]
..just to add, my CP version is:

Version4603b0c3b96e81d6d635e3cc3cab2ef9352d17c7 (1/15/2015, 12:01:25 AM)Updatersource, masterID5496 Directories/usr/pbi/couchpotato-amd64/share/couchpotato/CouchPotatoServer/usr/pbi/couchpotato-amd64/etc/couchpotatoStartup Args--config_file/usr/pbi/couchpotato-amd64/etc/couchpotato/config.ini--data_dir/usr/pbi/couchpotato-amd64/etc/couchpotato--pid_file/usr/pbi/couchpotato-amd64/etc/couchpotato/couchpotato.pid--daemonNamespace(config_file='/usr/pbi/couchpotato-amd64/etc/couchpotato/config.ini', console_log=False, daemon=True, data_dir='/usr/pbi/couchpotato-amd64/etc/couchpotato', debug=False, pid_file='/usr/pbi/couchpotato-amd64/etc/couchpotato/couchpotato.pid', quiet=False)

**Comment by bigmascot** [comment]
Ok, I figured it out.. despite thinking this was a problem relating solely to unrar, it turns out it was a permissions issue - which I gather is pretty common.  Just wanted to post here for anyone else suffering the same frustrating problem!

I'm on FreeNAS so from a terminal I entered the following:
chmod -R 777 /mnt/media/jails/couchpotato_1/usr/pbi/couchpotato-amd64/share/couchpotato/CouchPotatoServer/libs/unrar2/

I'm not sure why the permissions were incorrect as it has been running ok before, but at least it's fixed.  Happy days :)
